shared/test/test_deliverer.rb

```diff
@@ -0,0 +1,48 @@
+require_relative 'test_helper'
+require 'cdo/poste'
+
+class FakeSmtp
+  def send_message(message, from_address, *to_addresses)
+    @message = message
+    @from_address = from_address
+    @to_addresses = to_addresses
+  end
+end
+
+class DelivererTest < Minitest::Test
+  def setup
+    @fake_smtp = FakeSmtp.new
+    Deliverer.any_instance.stubs(:reset_connection).returns(@fake_smtp)
+    @deliverer = Deliverer.new({})
+  end
+
+  def test_deliverer_render
+    template = @deliverer.load_template("dashboard")
+    header, html, text = template.render
+    assert_equal "Code.org <noreply@code.org>", header["from"]
+    refute_nil html
+    assert_nil text
+  end
+
+  def test_deliverer_send
+    email = "test@deliverer.send"
+    contact = Poste2.create_recipient(email, {ip_address: '5.6.7.8.'})
+
+    # Sequel doesn't have a "find or create by", so we implement it manually
+    message = POSTE_DB[:poste_messages].where(name: "dashboard").first
+    message_id = message.nil? ? POSTE_DB[:poste_messages].insert({name: "dashboard"}) : message[:id]
```

Comment:
TIL!  Looks like it [only exists if you're using Sequel models](https://www.rubydoc.info/github/jeremyevans/sequel/Sequel%2FModel%2FClassMethods:find_or_create), which we don't actually do.